sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -251,30 +258,124 @@ export class AssessmentControllerService {
-   * @param identifier identifier which is used to create dynamic data by appending grant code
+   * @param identifier identifier which is used to create dynamic data by appending award code
    * to it.
    * @returns dynamic award data of disbursement receipts of a given disbursement.
+   * @example
+   * finalAward: {
+   *    disbursementReceipt1cslf: 1000,
+   *    disbursementReceipt1csgp: 10001,
+   *    disbursementReceipt1bcsl: 1005,
+   *    disbursementReceipt1bcag: 1006,
+   *    disbursementReceipt1bgpd: 1007,
+   *    disbursementReceipt1sbsd: 1008,
+   *    disbursementReceipt2cslf: 1000,
+   *    disbursementReceipt2csgp: 10001,
+   *    disbursementReceipt2bcsl: 1005,
+   *    disbursementReceipt2bcag: 1006,
+   *    disbursementReceipt2bgpd: 1007,
+   *    disbursementReceipt2sbsd: 1008,
+      }
    */
   private populateDisbursementReceiptAwardValues(
     disbursementReceipts: DisbursementReceipt[],
-    disbursementScheduleId: number,
+    disbursementSchedule: DisbursementSchedule,
     identifier: string,
   ): Record<string, string | number> {
     const finalAward = {};
+    // Add all estimated awards to the list of receipts returned.
+    // Ensure that every estimated disbursement will be part of the summary.
+    disbursementSchedule.disbursementValues.forEach((award) => {
+      if (award.valueType === DisbursementValueType.BCTotalGrant) {
+        // BC Total grants are not part of the students grants and should not be part of the summary.
+        return;
+      }
+      const disbursementValueKey = this.createReceiptFullIdentifier(
+        identifier,
+        award.valueCode,
+      );
+      finalAward[disbursementValueKey] = null;
+    });
+    // Process the two expected receipts records for federal(FE) and the other for provincial(BC) awards.
     disbursementReceipts
       .filter(
-        (receipt) => receipt.disbursementSchedule.id === disbursementScheduleId,
+        (receipt) =>
+          receipt.disbursementSchedule.id === disbursementSchedule.id,
       )
       .forEach((receipt) => {
-        finalAward[`${identifier}Id`] = receipt.id;
+        // Check if a loan is part of the receipt (e.g. part-time provincial loans shouldn't be available).
+        const loanType = DisbursementReceiptService.getLoanAwardCode(
+          receipt.fundingType,
+          receipt.disbursementSchedule.studentAssessment.offering
+            .offeringIntensity,
+        );
+        if (loanType) {
+          // Add the loan to the list of awards returned.
+          const disbursementLoanKey = `${identifier}${loanType.toLowerCase()}`;
+          finalAward[disbursementLoanKey] = receipt.totalDisbursedAmount;
+        }
+        // Populate the receipt amount in the receipt awards.
+        // If an estimated disbursement award has no equivalent receipt its value will be left as null.
         receipt.disbursementReceiptValues.forEach((receiptValue) => {
```

Comment:
The iteration is only meaningful for funding type FE.  If we use that condition for BC we don't have to check for award type BCSG because the iteration wont execute.

Reply:
I will take a look on it 😉

Reply:
Changed.